packages/server/src/fhir/repo.ts

```diff
@@ -903,9 +903,9 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
       resource = resources[i];
       (resource.meta as Meta).compartment = this.getCompartments(resource);
 
-      await this.writeResource(conn, resource);
       await this.writeLookupTables(conn, resource, false);
     }
+    await this.batchWriteResources(conn, ...resources);
```

Comment:
Only nit is that this creating a new array with a lot of new indices, though it's probably negligible on perf